### ISSUE TITLE
GPU usage in Activity Monitor 

### ISSUE THREAD

**Issue by Vadim170** [ggerganov/llama.cpp#1878]
Is it normal that I don't see GPU usage in the Mac OS Activity Monitor when running Metal-based computations?

**Comment by ianscrivener** [comment]
I have an M2 MacbookPro and see GPU % in the 90s when running llama.cpp

this is my helper script that I **_source_** to fetch and build the latest llama.cpp code

```
conda activate gg.3.9.16
git pull
git status
make clear
LLAMA_METAL=1 make -j   
```

TBH: I haven't had much luck building with CMake...

**Comment by Vadim170** [comment]
@ianscrivener 
I tried to follow your instructions, there were no changes. I also tried to build via cmake:
```
mkdir build-metal
cd build-metal
cmake -DLLAMA_METAL=ENABLED ..
cmake -build . --configuration release
```
However, I noticed that only the CPU is being used and there is no GPU-related activity. The repository is exactly the latest version, the project was cleaned before the build.
<img width="541" alt="image" src="https://github.com/ggerganov/llama.cpp/assets/38509064/ca8e4048-b35b-43d0-b47e-637d016ee2d8">
